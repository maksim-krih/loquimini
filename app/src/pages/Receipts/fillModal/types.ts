export interface IProps {
  visible: boolean;
  receiptId: string;
  onCancel: () => void;
}