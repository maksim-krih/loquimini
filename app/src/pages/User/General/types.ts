export interface IProps {
  isCreate?: boolean;
}
