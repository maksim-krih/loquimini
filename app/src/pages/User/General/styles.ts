import {createUseStyles} from 'react-jss'

export const useStyles = createUseStyles({
  container: {
    
  },
  actionButtons: {
    marginBottom: 10
  }
})