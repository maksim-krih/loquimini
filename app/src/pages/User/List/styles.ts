import {createUseStyles} from 'react-jss'

export const useStyles = createUseStyles({
  container: {
    
  },
  actionButtons: {
    marginBottom: 10
  },
  gridButton: {
    padding: 0,
    width: 34,
    borderRadius: 5
  }
})