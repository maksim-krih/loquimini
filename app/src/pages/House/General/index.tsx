import { FC, useEffect, useState } from "react";
import { IProps } from "./types";
import { useStyles } from "./styles";
import { CreateHouse, House, User } from "../../../services/types";
import Api from "../../../services";
import { Button, Form, Input, InputNumber, Select, Typography } from "antd";
import { useHistory, useParams } from "react-router-dom";
import { RouterPaths, validateMessages } from "../../../consts";
import { HouseType, HouseTypeSelect } from "../../../enums";
import { MinusCircleOutlined, PlusOutlined } from "@ant-design/icons";
import { useToggle } from "ahooks";

const FormItem = Form.Item;
const { Title } = Typography;

const General: FC<IProps> = (props: IProps) => {
  const { isCreate } = props;
  const classes = useStyles();
  const history = useHistory();
  const { id } = useParams<{id: string}>();
  const [form] = Form.useForm();
  const [users, setUsers] = useState<Array<User>>([]);
  const [data, setData] = useState<House>(Object);
  const [houseType, setHouseType] = useState(HouseType.Private);
  const [loading, setLoading] = useState(false);
  const [isEdit, { toggle }] = useToggle(false);
  const readonly = !isCreate && !isEdit;

  useEffect(() => {
    getUsers();

    if (!isCreate) {
      getHouse();
    }
  }, []);

  const getUsers = () => {
    setLoading(true);

    Api.User.getAll()
    .then((response: Array<User>) => {
      setUsers(response);
    })
    .finally(() => {
      setLoading(false);
    });
  };

  const getHouse = () => {
    Api.House.getById(id)
    .then((response: House) => {
      setData(response);
      setHouseType(response.type);
      form.resetFields();
    })
  };

  const onCancel = () => {
    if (isCreate || !isEdit)
    {
      history.goBack();
    }
    else {
      form.resetFields();
      toggle();
    }
  }

  const onFinish = (values: any) => {
    if(isCreate) {
      const model = values as CreateHouse;

      Api.House.create(model)
      .then(() => {
        history.push(RouterPaths.HouseList);
      });
    }
    else {
      const model = values as House;
      model.id = data.id;

      Api.House.update(model)
      .then(() => {
        history.push(RouterPaths.HouseList);
      });
    }
  };

  const onFinishFailed = (errorInfo: any) => {
    console.log('Failed:', errorInfo);
  };

  const onTypeChange = (value: HouseType) => {
    setHouseType(value);
  };

  return (
    <div className={classes.container}>
      {isCreate ? (
        <Button onClick={() => form.submit()}>
          Create
        </Button>
      ) : (
        <Button onClick={() => isEdit ? form.submit() : toggle()}>
          Edit
        </Button>
      )}
      <Button onClick={onCancel}>
        Cancel
      </Button>
      <Form
        labelCol={{ span: 3 }}
        wrapperCol={{ span: 10 }}
        form={form}
        name="house"
        validateMessages={validateMessages}
        initialValues={data.id ? data : {
          type: HouseType.Private
        }}
        onFinish={onFinish}
        onFinishFailed={onFinishFailed}
      >
        <FormItem
          label="Street"
          name="street"
          rules={[{ required: true }]}
        >
          <Input disabled={readonly}/>
        </FormItem>
    
        <FormItem
          label="Number"
          name="number"
          rules={[{ required: true }]}
        >
          <Input disabled={readonly}/>
        </FormItem>

        <FormItem
          label="Type"
          name="type"
          rules={[{ required: true }]}
        >
          <Select
            options={HouseTypeSelect}
            onChange={onTypeChange}
            disabled={readonly}
          />
        </FormItem>
        {houseType === HouseType.Private ? (
          <>
            <FormItem
              label="User"
              name="userId"
              rules={[{ required: true }]}
            >
              <Select
                options={users.map(x => ({
                  value: x.id,
                  label: x.email
                }))}
                disabled={readonly}
              />
            </FormItem>
            <FormItem
              label="Area"
              name={["info", "area"]}
              rules={[{ required: true }]}
            >
              <InputNumber disabled={readonly} />
            </FormItem>
            <Title level={5}>Default Indicators</Title>
            <FormItem
              label="Cold Water"
              name="coldWater"
            >
              <InputNumber disabled={readonly} />
            </FormItem>
            <FormItem
              label="Hot Water"
              name="hotWater"
            >
              <InputNumber disabled={readonly} />
            </FormItem>
            <FormItem
              label="Electricity"
              name="electricity"
            >
              <InputNumber disabled={readonly} />
            </FormItem>
            <FormItem
              label="Gas"
              name="gas"
            >
              <InputNumber disabled={readonly} />
            </FormItem>
          </>
        ) : (
          <>
            <Title level={4}>Flats</Title>
            <Form.List name="flats">
              {(fields, { add, remove }) => (
                <>
                  {fields.map(({ key, name, fieldKey, ...restField }, index) => (
                    <>
                      <Title level={5}>#{index + 1}</Title>
                      <MinusCircleOutlined onClick={() => remove(name)} />
                      <Form.Item
                        {...restField}
                        name={[name, 'number']}
                        fieldKey={[fieldKey, 'number']}
                        label="Number"
                        rules={[{ required: true }]}
                      >
                        <Input disabled={readonly} />
                      </Form.Item>
                      <Form.Item
                        {...restField}
                        name={[name, 'userId']}
                        fieldKey={[fieldKey, 'userId']}
                        label="User"
                        rules={[{ required: true }]}
                      >
                        <Select
                          options={users.map(x => ({
                            value: x.id,
                            label: x.email
                          }))}
                          disabled={readonly}
                        />
                      </Form.Item>
                      <Form.Item
                        {...restField}
                        name={[name, 'info', 'area']}
                        fieldKey={[fieldKey, 'info', 'area']}
                        label="Area"
                        rules={[{ required: true }]}
                      >
                        <InputNumber disabled={readonly} />
                      </Form.Item>
                      <Title level={5}>Default Indicators</Title>
                      <FormItem
                        label="Cold Water"
                        name="coldWater"
                      >
                        <InputNumber disabled={readonly} />
                      </FormItem>
                      <FormItem
                        label="Hot Water"
                        name="hotWater"
                      >
                        <InputNumber disabled={readonly} />
                      </FormItem>
                      <FormItem
                        label="Electricity"
                        name="electricity"
                      >
                        <InputNumber disabled={readonly} />
                      </FormItem>
                      <FormItem
                        label="Gas"
                        name="gas"
                      >
                        <InputNumber disabled={readonly} />
                      </FormItem>
                    </>
                  ))}
                  <Form.Item>
                    <Button type="dashed" onClick={() => add()} block icon={<PlusOutlined />}>
                      Add field
                    </Button>
                  </Form.Item>
                </>
              )}
            </Form.List>
          </>
        )}
      </Form>
    </div>
  );
}

export default General;