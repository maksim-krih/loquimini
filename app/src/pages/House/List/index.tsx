import { FC, useEffect, useState } from "react";
import { IProps } from "./types";
import { useStyles } from "./styles";
import { GridPager, GridRequest, House, User } from "../../../services/types";
import Api from "../../../services";
import { Button, Table, TablePaginationConfig, Typography } from "antd";
import { useHistory } from "react-router";
import { DefaultGridRequest, DefaultPager, RouterPaths } from "../../../consts";

const { Link } = Typography;

const List: FC<IProps> = (props: IProps) => {
  const classes = useStyles();
  const history = useHistory();
  const [data, setData] = useState([]);
  const [pager, setPager] = useState(DefaultPager);
  const [loading, setLoading] = useState(false);

  const columns = [
    {
      title: 'Street',
      dataIndex: 'street',
    },
    {
      title: 'Number',
      dataIndex: 'number',
    },
    {
      title: '',
      dataIndex: '',
      render: (_: any, record: House) => (
        <Link onClick={(e) => {
          e.stopPropagation();
          onDelete(record.id)
        }}>
          Delete
        </Link>
      )
    },
  ];

  useEffect(() => {
    getHousesData(pager);
  }, []);

  const getHousesData = (pager: GridPager) => {
    setLoading(true);

    const request = DefaultGridRequest(pager);

    Api.House.getAllGrid(request)
    .then((response: any) => {
      setData(response.data);
      setPager({...pager, total: response.total});
    })
    .catch(e => {
      
    })
    .finally(() => {
      setLoading(false);
    });
  };

  const handleTableChange = (pagination: TablePaginationConfig) => {
    setPager({...pager, current: pagination.current!});
    getHousesData({...pager, current: pagination.current!});
  };

  const onCreate = () => {
    history.push(RouterPaths.CreateHouse);
  }

  const onDelete = (id: string) => {
    setLoading(true);

    Api.House.deleteById(id)
    .then((response: any) => {
      if (response) {
        setPager({...pager, current: 1});
        getHousesData({...pager, current: 1});
      }
    })
    .catch(e => {
    
    })
    .finally(() => {
      setLoading(false);
    });
  }

  return (
    <div className={classes.container}>
      <Button onClick={onCreate}>
        Create
      </Button>
      <Table
        columns={columns}
        rowKey={(record: House) => record.id}
        dataSource={data}
        pagination={pager}
        loading={loading}
        onChange={handleTableChange}
        onRow={(record: House) => {
          return {
            onClick: () => history.push(RouterPaths.GeneralHouse(record.id)),
          };
        }}
      />
    </div>
  );
}

export default List;