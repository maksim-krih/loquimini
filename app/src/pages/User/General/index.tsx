import { FC, useEffect, useState } from "react";
import { IProps } from "./types";
import { useStyles } from "./styles";
import { CreateUser } from "../../../services/types";
import Api from "../../../services";
import { Button, Form, Input } from "antd";
import { useHistory } from "react-router-dom";
import { RouterPaths, validateMessages } from "../../../consts";

const FormItem = Form.Item;
const { Password } = Input;

const General: FC<IProps> = (props: IProps) => {
  const classes = useStyles();
  const history = useHistory();
  const [form] = Form.useForm();
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    getUsersData();
  }, []);

  const getUsersData = () => {
  };

  const onCancel = () => {
    history.goBack()
  }

  const onFinish = (values: CreateUser) => {
    values.roles = ["User"];
    Api.User.create(values)
    .then(() => {
      history.push(RouterPaths.UserList);
    });
  };

  const onFinishFailed = (errorInfo: any) => {
    console.log('Failed:', errorInfo);
  };

  const renderCreate = (
    <div>
      <Button onClick={() => form.submit()}>
        Create
      </Button>
      <Button onClick={onCancel}>
        Cancel
      </Button>
      <Form
        labelCol={{ span: 3 }}
        wrapperCol={{ span: 10 }}
        form={form}
        name="user"
        validateMessages={validateMessages}
        onFinish={onFinish}
        onFinishFailed={onFinishFailed}
      >
        <FormItem
          label="First Name"
          name="firstName"
          rules={[{ required: true }]}
        >
          <Input />
        </FormItem>
    
        <FormItem
          label="Last Name"
          name="lastName"
          rules={[{ required: true }]}
        >
          <Input />
        </FormItem>

        <FormItem
          label="Email"
          name="email"
          rules={[{ required: true }]}
        >
          <Input type="email" />
        </FormItem>

        <FormItem
          label="Password"
          name="password"
          rules={[{ required: true }]}
        >
          <Password />
        </FormItem>
      </Form>
    </div>
  );

  return (
    <div className={classes.container}>
      {props.isCreate ? renderCreate
      : (
        <div></div>
      )}
    </div>
  );
}

export default General;