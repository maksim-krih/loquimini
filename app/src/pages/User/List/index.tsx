import { FC, useEffect, useState } from "react";
import { IProps } from "./types";
import { useStyles } from "./styles";
import { GridRequest, User } from "../../../services/types";
import Api from "../../../services";
import { Button, Table } from "antd";
import { useHistory } from "react-router";
import { RouterPaths } from "../../../consts";

const columns = [
  {
    title: 'Name',
    dataIndex: 'userName',
    render: (text: string, record: User)=> `${record.firstName} ${record.lastName}`
  },
  {
    title: 'Email',
    dataIndex: 'email'
  },
];

const List: FC<IProps> = (props: IProps) => {
  const classes = useStyles();
  const history = useHistory();
  const [data, setData] = useState([]);
  const [pager, setPager] = useState({
    current: 1,
    pageSize: 10
  });
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    getUsersData();
  }, []);

  const getUsersData = () => {
    setLoading(true);

    const request: GridRequest = {
      filters: [],
      pager,
      search: {
        fields: [],
        value: ""
      },
      sorter: []
    };

    Api.User.getAllGrid(request)
    .then((response: any) => {
      setData(response.data);
    })
    .finally(() => {
      setLoading(false);
    });
  };

  const handleTableChange = (pagination: any) => {
    setPager(pagination);
    getUsersData();
  };

  const onCreate = () => {
    history.push(RouterPaths.CreateUser);
  }

  return (
    <div className={classes.container}>
      <Button onClick={onCreate}>
        Create
      </Button>
      <Table
        columns={columns}
        rowKey={(record: User) => record.id}
        dataSource={data}
        pagination={pager}
        loading={loading}
        onChange={handleTableChange}
      />
    </div>
  );
}

export default List;