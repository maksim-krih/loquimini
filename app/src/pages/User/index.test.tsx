import General from "./General";
import List from "./List";
import { shallow } from "enzyme";
import { Route } from "react-router-dom";
import { IProps as IGeneralProps } from "./General/types";

const setUpGeneral = (props: IGeneralProps) => shallow(
  <Route path="/user/:id" >
    <General {...props} />
  </Route>
);
const setUpList = () => shallow(<List />);

describe("User", () => {
  describe("Edit", () => {
    it("should render Edit Component", () => {
      const component = setUpGeneral({
        isCreate: false
      });
      expect(component).toMatchSnapshot();
    });
  });

  describe("Create", () => {
    it("should render Create Component", () => {
      const component = setUpGeneral({
        isCreate: true
      });
      expect(component).toMatchSnapshot();
    });
  });

  describe("List", () => {
    it("should render List Component", () => {
      const component = setUpList();
      expect(component).toMatchSnapshot();
    });
  });
});