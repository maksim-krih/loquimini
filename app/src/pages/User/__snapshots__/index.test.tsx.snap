// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Create should render Create Component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`User Edit should render Edit Component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`User List should render List Component 1`] = `
<div
  className="container-0-2-1"
>
  <div
    className="actionButtons-0-2-2"
  >
    <Button
      onClick={[Function]}
      style={
        Object {
          "borderRadius": 5,
        }
      }
      type="primary"
    >
      Create
    </Button>
  </div>
  <Table
    columns={
      Array [
        Object {
          "dataIndex": "userName",
          "search": true,
          "sorter": true,
          "title": "Name",
        },
        Object {
          "dataIndex": "email",
          "search": true,
          "sorter": true,
          "title": "Email",
        },
        Object {
          "dataIndex": "",
          "render": [Function],
          "title": "",
        },
      ]
    }
    data={Array []}
    getData={[Function]}
    loading={false}
    onRow={[Function]}
    pager={
      Object {
        "current": 1,
        "pageSize": 10,
        "total": 0,
      }
    }
    setPager={[Function]}
  />
</div>
`;
